q2_fmt/_peds.py

```diff
@@ -415,52 +416,78 @@ def sample_pprs(table: pd.DataFrame, metadata: qiime2.Metadata,
                        subject_column, 'categorical')
     _check_duplicate_subject_timepoint(subject_series, metadata_df,
                                        subject_column, time_column)
-
-    peds_df = pd.DataFrame(columns=['id', 'measure',
-                                    'transfered_baseline_features',
-                                    'total_baseline_features', 'baseline',
-                                    'subject', 'group'])
     baseline_metadata = metadata_df.join(used_references)
-    peds_df = _compute_peds(peds_df=peds_df, peds_type='PPRS',
-                            peds_time=np.nan, reference_series=used_references,
-                            table=table, metadata=baseline_metadata,
-                            time_column=time_column,
-                            subject_column=subject_column,
-                            reference_column=used_references.name)
-
-    peds_type = 'PPRS'
-    transfered = "transfered_baseline_features"
-    total = 'total_baseline_features'
-    ref = 'baseline'
-    measure_description = ('Proportional Persistence of Recipient'
-                           ' Strains')
+
+    numerator_df = pd.DataFrame([], index=table.index)
+    denominator_df = pd.DataFrame([], index=table.index)
+    if num_resamples == 0:
+        # This will allow the next block of code to run one with
+        # no samping depth
+        num_resamples = 1
+    for x in range(num_resamples):
+        peds_df = pd.DataFrame(columns=['id',
+                                        'transfered_baseline_features',
+                                        'total_baseline_features',
+                                        'baseline', 'subject',
+                                        'group'])
+        if sampling_depth:
+            table = _subsample(table, sampling_depth)
+        peds_df = _compute_peds(peds_df=peds_df, peds_type='PPRS',
+                                peds_time=np.nan,
+                                reference_series=used_references,
+                                table=table, metadata=baseline_metadata,
+                                time_column=time_column,
+                                subject_column=subject_column,
+                                reference_column=used_references.name)
+        # Set the index for pd matching when concating and summing
+        peds_df = peds_df.set_index('id')
+        numerator_df = pd.concat([peds_df['transfered_baseline_features'],
+                                  numerator_df],
+                                 axis=1)
+        denominator_df = pd.concat([peds_df['total_baseline_features'],
+                                    denominator_df],
+                                   axis=1)
```

Comment:
Maybe use a list of series and call concat once at the end of all of the iterations.